guides/source/action_view_overview.md

```diff
@@ -219,183 +205,137 @@ One way to use partials is to treat them as the equivalent of subroutines; a way
-  #    </div>
-%>
 ```
 
-Pattern matching assignment also supports variable renaming:
-
-```ruby
-local_assigns => { product: record }
-product             # => "#<Product:0x0000000109ec5d10>"
-record              # => "#<Product:0x0000000109ec5d10>"
-product == record   # => true
-```
-
-Since `local_assigns` returns a `Hash` instance, you can conditionally read a variable, then fall back to a default value when the key isn't part of the `locals:` options:
+Another example from Active Storage's `_blob.html.erb`. This one sets the size based on whether `in_gallery` local variable is set when rendering the partial that contains this line:
 
 ```html+erb
-<%# app/views/products/_product.html.erb %>
-
-<% local_assigns.fetch(:related_products, []).each do |related_product| %>
-  <%# ... %>
-<% end %>
+<%= image_tag blob.representation(resize_to_limit: local_assigns[:in_gallery] ? [ 800, 600 ] : [ 1024, 768 ]) %>
 ```
 
-Combining Ruby 3.1's pattern matching assignment with calls to [Hash#with_defaults](https://api.rubyonrails.org/classes/Hash.html#method-i-with_defaults) enables compact partial-local default variable assignments:
-
-```html+erb
-<%# app/views/products/_product.html.erb %>
-
-<% local_assigns.with_defaults(related_products: []) => { product:, related_products: } %>
-
-<%= tag.div id: dom_id(product) do %>
-  <h1><%= product.name %></h1>
-
-  <% related_products.each do |related_product| %>
-    <%# ... %>
-  <% end %>
-<% end %>
-```
-
-INFO: By default, partials will accept any `locals` as keyword arguments. To define what `locals` a partial accepts, use a `locals:` magic comment. To learn more, read about [Strict Locals](#strict-locals).
-
-[local_assigns]: https://api.rubyonrails.org/classes/ActionView/Template.html#method-i-local_assigns
-
 ### `render` without `partial` and `locals` Options
 
-In the above example, `render` takes 2 options: `partial` and `locals`. But if
-these are the only options you want to pass, you can skip using these options.
+In the above examples, `render` takes 2 options: `partial` and `locals`. But if
+these are the only options you need to use, you can skip the keys, `partial` and `locals` and specify the values only.
+
 For example, instead of:
 
 ```erb
 <%= render partial: "product", locals: { product: @product } %>
 ```
 
-You can also do:
+You can write:
 
 ```erb
 <%= render "product", product: @product %>
 ```
 
+You can also use this shorthand based on conventions:
+
+```erb
+<%= render @product %>
+```
+
+This will look for a partial named `_product.html.erb` in `app/views/products/`, as well as pass a local named `product` set to the value `@product`.
+
+
 ### The `as` and `object` Options
 
-By default `ActionView::Partials::PartialRenderer` has its object in a local variable with the same name as the template. So, given:
+By default, objects passed to the template are in a local variable with the same name as the template. So, given:
 
 ```erb
 <%= render partial: "product" %>
```

Comment:
I'm wondering if we should change this to just:

```suggestion
<%= render @product %>
```

i.e., the shorter example using the convention explained on the previous section... this removes any "inference" from the example and makes it simpler / more explicit.

Reply:
Yes because we *just* explained it, I think the shorthand is cleaner in this example.